sources/packages/backend/apps/queue-consumers/src/processors/schedulers/institution-integration/ier12-integration/_tests_/e2e/ier12-integration.scheduler.e2e-spec.ts

```diff
@@ -0,0 +1,394 @@
+    );
+  });
+
+  it("Should generate an IER12 file with one record for a married student (mononymous) with dependents, applications still in assessment, and one pending disbursement.", async () => {
+    // Arrange
+    const testInputData = {
+      student: JANE_MONONYMOUS_FROM_OTHER_COUNTRY,
+      application: {
+        applicationNumber: defaultApplicationNumber,
+        studentNumber: "12345679",
+        relationshipStatus: RelationshipStatus.Married,
+        applicationStatus: ApplicationStatus.Assessment,
+        applicationStatusUpdatedOn: undefined,
+      },
+      assessment: {
+        triggerType: AssessmentTriggerType.OriginalAssessment,
+        assessmentDate: undefined,
+        workflowData: WORKFLOW_DATA_MARRIED_WITH_DEPENDENTS,
+        assessmentData: ASSESSMENT_DATA_MARRIED,
+        disbursementSchedules: [
+          {
+            coeStatus: COEStatus.completed,
+            disbursementScheduleStatus: DisbursementScheduleStatus.Pending,
+            disbursementDate: undefined,
+            updatedAt: undefined,
+            dateSent: undefined,
+            disbursementValues: AWARDS_SINGLE_DISBURSEMENT,
+          },
+        ],
+      },
+      educationProgram:
+        PROGRAM_UNDERGRADUATE_CERTIFICATE_WITHOUT_INSTITUTION_PROGRAM_CODE,
+      offering: OFFERING_FULL_TIME,
+    };
+    const application = await saveIER12TestInputData(db, testInputData, {
+      programYearPrefix: sharedProgramYearPrefix,
+      submittedDate: referenceSubmissionDate,
+    });
+
+    // Queued job.
+    const job = createIER12SchedulerJobMock(
+      application.currentAssessment.assessmentDate,
+    );
+
+    // Act
+    const ier12Results = await processor.processIER12File(job);
+    db.application.createQueryBuilder();
+
+    // Assert
+    const uploadedFile = getUploadedFile(sftpClientMock);
+    // Assert process result.
+    expect(ier12Results).toBeDefined();
+    // File timestamp.
+    const [timestampResult] = getFileNameAsCurrentTimestampMock.mock.results;
+    expect(isValidFileTimestamp(timestampResult.value)).toBe(true);
+    expect(ier12Results).toStrictEqual([
+      getSuccessSummaryMessages(timestampResult.value),
+    ]);
+    // Assert file output.
+    expect(uploadedFile.fileLines?.length).toBe(1);
+    const [line1] = uploadedFile.fileLines;
+    const [firstDisbursement] =
+      application.currentAssessment.disbursementSchedules;
+    const assessmentId = numberToText(application.currentAssessment.id);
```

Comment:
To avoid using the code that would also potentially be validated. If we introduce an error in the `StringBuilder` we may end up not catching it in the E2Es. Same reason that we reuse code from the application but with caution, try to keep the E2E assertion validations as "raw" as possible.

Reply:
File moved.